codice/ddf
# DDF-4901 Provide an extension point for query forms in Intrigue

#### What does this PR do?
Provide an extension point for query forms in Intrigue, these forms currently include: Text Search, Basic Search and Advanced Search. Allow downstream projects to add custom query forms.
#### Who is reviewing it? 
@hayleynorton @willwill96 @zta6 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler 
@djblue 

#### How should this be tested?
- build / run
- open a new search, verify you can switch between all forms and make searches without an error
#### Any background context you want to provide?
#### What are the relevant tickets?
For GH Issues:
Fixes: #4901 

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
